frontend/src/metabase/styled-components/theme/theme.ts

```diff
@@ -0,0 +1,47 @@
+import { get } from "lodash";
+
+import type { MetabaseComponentTheme } from "embedding-sdk";
+import { useMantineTheme } from "metabase/ui";
+
+// https://www.raygesualdo.com/posts/flattening-object-keys-with-typescript-types/
+type FlattenObjectKeys<
+  T extends Record<string, unknown>,
+  Key = keyof T,
+> = Key extends string
+  ? T[Key] extends Record<string, unknown>
+    ? `${Key}.${FlattenObjectKeys<T[Key]>}`
+    : `${Key}`
+  : never;
+
+type MetabaseComponentThemeKey = FlattenObjectKeys<MetabaseComponentTheme>;
```

Comment:
This gives us a little bit of type safety when we're trying to get values from the `MetabaseComponentTheme` - try it out for yourself and put a key that isn't in the theme object, and you should get a TS error

Reply:
neat!